src/index.ts

```diff
@@ -201,13 +201,14 @@ class SmartQueue {
       return;
     }
 
-    debug('Executing queue item', nextItem.item.id);
+    this.heat();
 
-    const startTime = Date.now();
+    debug('Executing queue item', nextItem.item.id);
 
     try {
-      const data = await nextItem.item.request(retryFn);
-
+      const requestPromise = nextItem.item.request(retryFn);
+      this.execute();
```

Comment:
Am I understand correctly that we are calling `execute` recursively? But why?

Reply:
It is the same call as before, just not after the request is done.
Notice we don't wait for the execute return value.
This is required to make sure that if there are more pending requests, they will be waited for the appropriate coolddown time, and if there are no pendings, mark the `pending` property as false